src/generators/legacy-json/utils/buildSection.mjs

```diff
@@ -0,0 +1,385 @@
+ */
+function createSection(entry, head) {
+  const text = transformNodesToString(head.children);
+  return {
+    textRaw: text,
+    name: head.data.name,
+    type: head.data.type,
+    meta: createMeta(entry),
+    introduced_in: entry.introduced_in,
+  };
+}
+
+/**
+ * Find name, type, default, desc properties
+ * @param {import('mdast').ListItem} child
+ * @param {import('../types.d.ts').HierarchizedEntry} entry
+ * @returns {import('../types.d.ts').List}
+ */
+function parseListItem(child, entry) {
+  /**
+   * @type {import('../types.d.ts').List}
+   */
+  const current = {};
+
+  current.textRaw = child.children
+    .filter(node => node.type !== 'list')
+    .map(node =>
+      entry.rawContent.slice(
+        node.position.start.offset,
+        node.position.end.offset
+      )
+    )
+    .join('')
+    .replace(/\s+/g, ' ')
+    .replaceAll(/<!--.*?-->/gs, '');
+
+  if (!current.textRaw) {
+    throw new Error(`Empty list item: ${JSON.stringify(child)}`);
+  }
+
+  let text = current.textRaw;
+
+  // Extract name
+  if (RETURN_EXPRESSION.test(text)) {
+    current.name = 'return';
+    text = text.replace(RETURN_EXPRESSION, '');
+  } else {
+    const [, name] = text.match(NAME_EXPRESSION) || [];
+    if (name) {
+      current.name = name;
+      text = text.replace(NAME_EXPRESSION, '');
+    }
+  }
+
+  // Extract type (if provided)
+  const [, type] = text.match(TYPE_EXPRESSION) || [];
+  if (type) {
+    current.type = type;
+    text = text.replace(TYPE_EXPRESSION, '');
+  }
+
+  // Remove leading hyphens
+  text = text.replace(LEADING_HYPHEN, '');
+
+  // Extract default value (if exists)
+  const [, defaultValue] = text.match(DEFAULT_EXPRESSION) || [];
+  if (defaultValue) {
+    current.default = defaultValue.replace(/\.$/, '');
+    text = text.replace(DEFAULT_EXPRESSION, '');
+  }
+
+  // Add remaining text to the desc
+  if (text) {
+    current.desc = text;
+  }
+
+  const options = child.children.find(child => child.type === 'list');
+  if (options) {
+    current.options = options.children.map(child =>
+      parseListItem(child, entry)
+    );
+  }
+
+  return current;
+}
+
+/**
+ * @param {import('../types.d.ts').HierarchizedEntry} entry
+ * @param {import('../types.d.ts').Section} parentSection
+ */
+function handleEntry(entry, parentSection) {
+  // Clone the children so we don't mess with any other generators
+  let [headingNode, ...nodes] = structuredClone(entry.content.children);
+
+  /**
+   * Grabs stability number & text and adds it to the section
+   * @param {import('../types.d.ts').Section} section
+   */
+  const parseStability = section => {
+    let needsCompressing = false;
+
+    // Remove metadata not directly the inferable from the markdown
+    nodes.forEach((node, i) => {
+      if (
+        node.type === 'blockquote' &&
+        node.children.length === 1 &&
+        node.children[0].type === 'paragraph' &&
+        nodes.slice(0, i).every(node => node.type === 'list')
```

Comment:
Wouldn't it be better to simply extract the nodes that are listed and ignore others? Also, we already have helpers for stability index, I don't think any of this needs to be done. Afaik the metadata of the node should already have everything about stability. Meaning we don't need any of this code (IMO)

See: https://github.com/nodejs/api-docs-tooling/blob/main/src/metadata.mjs#L152

Reply:
We need to delete the stability node from the nodes